code/WordpressImporter.php

```diff
@@ -0,0 +1,611 @@
+			} else {
+				$this->tags[$title] = BlogTag::get()->filter("Title", $title)
+					->filter("BlogID", $this->getBlog()->ID)
+					->first();
+			}
+		}
+	}
+
+
+
+	/**
+	 * Import the blog posts and link them up to any attachments.
+	 *
+	 * @param $element SimpleXMLElement
+	**/
+	public function importBlogPost() {
+		foreach($this->simpleXml->channel->children()->item as $item) {
+			$content = $item->children($this->namespaces['content'])->encoded;
+			$excerpt = $item->children($this->namespaces['excerpt'])->encoded;
+			$post = $item->children($this->namespaces['wp']);
+
+			// Check for existing post
+			if($post->post_type == "post" && !array_key_exists((string) $post->post_id, $this->posts)) {
+				$blogPost = BlogPost::create();
+				$blogPost->Title = (string) $item->title;
+				$blogPost->MetaTitle = (string) $item->title;
+				$blogPost->MetaDescription = (string) $excerpt;
+				$blogPost->URLSegment = (string) $post->post_name;
+				$blogPost->setCastedField("PublishDate", $post->post_date_gmt);
+				$blogPost->Content = (string) $this->parseHTML($content);
+				$blogPost->WordpressID = (string) $post->post_id;
+				$blogPost->ParentID = $this->getBlog()->ID;
+
+				// Hook to update BlogPost
+				$this->extend("beforeImportBlogPost", $item, $blogPost);
+
+				if(!$this->isDryRun()) {
+					$blogPost->writeToStage("Stage");
+					if((string) $post->status == "publish") {
+						$blogPost->publish("Stage", "Live");
+					}
+				}
+
+				$this->posts[$blogPost->WordpressID] = $blogPost;
+				$this->addImportedObject("BlogPost", $blogPost);
+
+				// Add attachments
+				$this->addAttachments($item, $blogPost);
+			}
+		}
+	}
+
+
+
+	/**
+	 * Imports images and other uploaded files.
+	 *
+	 * @param $element SimpleXMLElement
+	**/
+	public function importAssets() {
+		if($this->getImportAssets()) {
+			foreach($this->simpleXml->channel->children()->item as $item) {
+				$attachment = $item->children($this->namespaces['wp']);
+
+				$url = (string) $attachment->attachment_url;
+				if($url) {
+					$split = explode("/", $url);
+					$filename = end($split);
+					$reversed = array_reverse($split);
+					if(count($reversed) > 2) {
+						$month = $reversed[1];
+						$year = $reversed[2];
+
+						$dir = Controller::join_links("Uploads", $year, $month);
+					} else {
+						$dir = Controller::join_links('Imported');
+					}
+
+					$folder = $dir;
+					if(!$this->isDryRun()) {
+						$folder = Folder::find_or_make($dir);
+						$folder = $folder->Filename;
+					}
+
+					$filename = Controller::join_links(Director::baseFolder(), $folder, $filename);
+					if(!file_exists($filename)) {
+						if(!$this->isDryRun()) {
+							// Lets get that image!
+							$source = fopen($url, 'r');
+							$dest = fopen($filename, 'w');
+							if($source && $dest) {
+								stream_copy_to_stream($source, $dest);
+							}
+							if($source) fclose($source);
+							if($dest) fclose($dest);
+							Filesystem::sync();
```

Comment:
I wonder if its worth creating the File object manually? Filesystem::sync() is scary and deprecated in ss4.